test/src/org/labkey/test/tests/mobileappstudy/TokenValidationTest.java

```diff
@@ -216,6 +217,35 @@ public void testResolveEnrollmentToken()
         assertEquals(STUDY_NAME03, resolveCmd.getStudyId());
     }
 
+    @Test
+    public void testMyStudiesCoordinatorRole()
+    {
+        goToProjectHome(PROJECT_NAME01);
+        SetupPage setupPage = new SetupPage(this);
+        TokenBatchesWebPart batchesWebPart = setupPage.getTokenBatchesWebPart();
+
+        // Test for Administrator
+        assertTrue(batchesWebPart.isNewBatchPresent());
+        assertTrue(batchesWebPart.isNewBatchEnabled());
+        setupPage.validateSubmitButtonDisabled();  // Submit button should be present
+
+        // Test for Reader
+        impersonateRole("Reader");
+        assertTrue(batchesWebPart.isNewBatchPresent());
+        assertFalse(batchesWebPart.isNewBatchEnabled());
+        assertFalse(setupPage.isSubmitButtonVisible());  // Submit button should NOT be present
+        stopImpersonating(false);
+
+        // Test for MyStudies Coordinator
+        impersonateRoles("Reader", "MyStudies Coordinator");
+        assertTrue(batchesWebPart.isNewBatchPresent());
+        assertTrue(batchesWebPart.isNewBatchEnabled());  // Should be able to create a new batch
+        assertFalse(setupPage.isSubmitButtonVisible());  // Submit button should NOT be present
```

Comment:
I must not be reading this correctly.  How is it useful for the "new batch" button to be enabled but the submit button not?  Is the popup not displayed at this point in the test?

Reply:
Ah, this is probably the submit button in the StudySetup panel.  Seems like that whole panel should be missing, but at least the Submit button shouldn't be there.  Never mind.

Reply:
Correct... this is just a proxy for "no administrator permissions." Everyone sees the webpart, but only administrators can interact with it. (Not exactly best practice to stick all the admin controls in a webpart... but that's how it works currently.)